craft_parts/plugins/rust_plugin.py

```diff
@@ -111,32 +118,84 @@ def get_build_snaps(self) -> Set[str]:
     @override
     def get_build_packages(self) -> Set[str]:
         """Return a set of required packages to install in the build environment."""
-        return {"gcc", "git"}
+        return {"curl", "gcc", "git", "pkg-config", "findutils"}
+
+    def _check_rustup(self) -> bool:
+        try:
+            rustup_version = subprocess.check_output(["rustup", "--version"])
+            return "rustup" in rustup_version.decode("utf-8")
+        except (subprocess.CalledProcessError, FileNotFoundError):
+            return False
+
+    def _get_setup_rustup(self, channel: str) -> List[str]:
+        return [
+            f"""\
+curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | \
+sh -s -- -y --no-modify-path --profile=minimal --default-toolchain {channel}
+"""
+        ]
 
     @override
     def get_build_environment(self) -> Dict[str, str]:
         """Return a dictionary with the environment to use in the build step."""
-        return {"PATH": "${HOME}/.cargo/bin:${PATH}"}
+        return {
+            "PATH": "${HOME}/.cargo/bin:${PATH}",
+        }
 
     @override
     def get_build_commands(self) -> List[str]:
         """Return a list of commands to run during the build step."""
         options = cast(RustPluginProperties, self._options)
 
-        install_command_list = [
-            "cargo",
-            "install",
-            "--locked",
-            "--path",
-            options.rust_path[0],  # TODO: support more than 1 path
-            "--root",
-            '"${CRAFT_PART_INSTALL}"',
-            "--force",
-        ]
+        rust_install_cmd: List[str] = []
+        rust_build_cmd: List[str] = []
+        config_cmd: List[str] = []
+
+        if options.rust_channel == "none":
+            rust_install_cmd = []
+        elif not self._check_rustup():
+            logger.info("Rustup not found, installing it")
+            rust_install_cmd = self._get_setup_rustup(options.rust_channel)
+        else:
+            logger.info("Switch rustup channel to %s", options.rust_channel)
+            rust_install_cmd = [f"rustup default {options.rust_channel}"]
 
         if options.rust_features:
-            install_command_list.extend(
-                ["--features", f"'{' '.join(options.rust_features)}'"]
+            features_string = " ".join(options.rust_features)
+            config_cmd.extend(["--features", f"'{features_string}'"])
```

Comment:
```suggestion
            config_cmd.extend(["--features", *options.rust_features])
```

Reply:
This suggested change does not seem correct.
`cargo` expect all the features to be a string:

`cargo build --features 'a b c'`

instead of:

`cargo build --features a b c`

Reply:
Ahh, you're right — I missed the `'`s. Sorry about that!